pkg/credential/credential.go

```diff
@@ -20,28 +20,61 @@ const (
 	defaultAzureGUID  = "37f13270-5c7a-56ff-9211-8426baaeaabd"
 )
 
-func GetTenantAzureClientCredentialsConfig(k8sClient kubernetes.Interface, cr providerv1alpha1.AzureConfig) (auth.ClientCredentialsConfig, error) {
+// GetOrganizationAzureClientCredentialsConfig returns ClientCredentialsConfig with the organization Tenant ID as auxiliary tenant id,
+// so that a Multi-Tenant Service Principal can work on the subscription linked to the organization Tenant ID.
+// It will fallback to a Single Tenant Service Principal when organization's credential secret contains Client ID and Client Secret.
+func GetOrganizationAzureClientCredentialsConfig(k8sClient kubernetes.Interface, cr providerv1alpha1.AzureConfig, gsClientCredentialsConfig auth.ClientCredentialsConfig) (auth.ClientCredentialsConfig, error) {
 	credential, err := k8sClient.CoreV1().Secrets(key.CredentialNamespace(cr)).Get(key.CredentialName(cr), apismetav1.GetOptions{})
 	if err != nil {
 		return auth.ClientCredentialsConfig{}, microerror.Mask(err)
 	}
 
-	clientID, err := valueFromSecret(credential, ClientIDKey)
+	clientID, _ := valueFromSecret(credential, ClientIDKey)
```

Comment:
We no longer fail if credentials are missing from credential secret. They being empty is the indicator that we should use multi tenant Service Principal.

Reply:
will it fail if multi tenant Service Principal is also not set or it will try to use single tenant set up again? (sorry if it's a stupid question!)

Reply:
Not stupid at all! The parameters are passed when installing, and the values come from the installations repository.

We are not checking if those parameters contain a value. We could add it.

Reply:
Fixed here @MartaD https://github.com/giantswarm/azure-operator/pull/840/commits/24800876858720cb4c3cb51e143f7783d332926f#diff-eeade751807b5e894d1f9a5b21390878R296

Reply:
We changed the mechanism to indicate that we should use multi tenant Service Principal: if the organization secret has a label, it's single tenant.

Reply:
I've changed it again :joy: . If the tenant id in the organization secret is the same as the tenant id passed as argument to azure-operator for authentication, then it's single tenant.